packages/dx-react-scheduler/src/plugins/all-day-panel.tsx

```diff
@@ -102,7 +107,10 @@ export class AllDayPanel extends React.PureComponent {
               const intervals = calculateAllDayDateIntervals(
                 appointments, startViewDate, endViewDate, excludedDays,
               );
-              const rects = tableRef && tableRef.querySelectorAll('th').length === viewCellsData[0].length ? calculateRectByDateIntervals(
+              const rects = tableRef
+              && (tableRef! as HTMLElement)
```

Comment:
Why is `as HTMLElement` used? `tableRef` is already of `HTMLElement` type.

Reply:
In the initial state, this is a `null`

Reply:
The `!` symbols tells that it is not `null` at that moment.

Reply:
My compiler execute error when I removed any of `!` or `as ...`